core/src/main/java/io/grpc/internal/AutoConfiguredLoadBalancerFactory.java

```diff
@@ -234,36 +268,15 @@ PolicySelection decideLoadBalancerProvider(
               getProviderOrThrow(
                   "round_robin", "received balancer addresses but grpclb runtime is missing"),
               backendAddrs, null);
-        } else {
-          return new PolicySelection(grpclbProvider, servers, null);
         }
+        return new PolicySelection(grpclbProvider, servers, null);
+      } else {
```

Comment:
It's confusing to have the else _and_ continue, since the if always returns. I'd suggest moving this out of the else and just put it after the if.